src/index.ts

```diff
@@ -148,6 +161,13 @@ export default class JSONFormatter {
     return Array.isArray(this.json);
   }
 
+  /*
+   * is this an array range?
+  */
+  private get isArrayRange(): boolean {
+    return this.isArray && this.hasKey && this.key.indexOf(RANGE_DELIMITER) >= 0;
```

Comment:
I wonder if we can somehow avoid depending on existence of a random string in the key. JSON keys can have anything in them!

Reply:
Yes, it is a legitimate request. Indeed, I wasn't completely sure of this technique.